File: week06/32/241008.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # Implement your solution here
    arr = [float('-inf')] * len(A)
    arr[0] = A[0]

    for i in range(1, len(A)):
        for j in range(1, 7):
            if i-j >=0:
                arr[i] = max(arr[i], arr[i-j]+A[i])

    return arr[len(A)-1]

File: week06/32/241008.cpp
// you can use includes, for example:
// #include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
    // Implement your solution here
    vector<int> dp(A.size(),0);
    dp[0] = A[0];

    for (int i=1; i < A.size(); ++i) {
        int m = dp[i-1];
        for (int j=2; j < 7; ++j) {
            if (i >= j) {
                m = max(m, dp[i-j]);
            }
        }
        dp[i] = m + A[i];
    }
    return dp[A.size()-1];
}